scripts/solve_network.py

```diff
@@ -130,16 +194,146 @@ def add_co2_sequestration_limit(n, limit=200):
         limit = float(o[o.find("seq") + 3 :]) * 1e6
         break
 
-    n.add(
+    if config["foresight"] == "perfect":
```

Comment:
```suggestion
    if not n.investment_periods.empty:
```
This pattern is more object related. Could perhaps be used elsewhere. Then the `config` argument can also be removed.